### DISCUSSION TITLE
How to watch validation on a form

### DISCUSSION THREAD

**Discussion by andyjamesn** [formkit/formkit#1292] [General]
I am creating a custom stepped form. I am not using the step form option as my UI is very custom.

The form us generated using a schema. 

The next step button is disabled and I am using live validation, what I'd like to be able to do is watch the validation and when all fields are valid enable the next step button. 

Is there any way to watch the validation?

**Comment by fenilli** [comment]
yes you can watch the valid state in the context, `toRef(getNode('formId').context.state, 'valid')`. you can also check the implementation for the multi-step here: https://github.com/formkit/formkit/blob/master/packages/addons/src/plugins/multiStep/multiStepPlugin.ts
